checks/grouping.py

```diff
@@ -0,0 +1,23 @@
+import numpy as np
+from astropy.table import Table
+from astropy.table import Table
+from astropy.coordinates import Angle, SkyCoord
+from gammapy.data import DataStore
+from gammapy.data import ObservationGroupAxis
+from gammapy.data import ObservationGroups
+from gammapy.datasets import gammapy_extra
+import pylab as pt
+pt.ion()
+
+dir=str(gammapy_extra.dir)+'/datasets/hess-crab4'
+data_store = DataStore.from_dir(dir)
+Observation_Table=data_store.obs_table
+alt = Angle([0, 30, 60, 90], 'degree')
+az = Angle([-90, 90, 270], 'degree')
+ntels = np.array([3, 4])
+list_obs_group_axis = [ObservationGroupAxis('ALT', alt, 'bin_edges'),
+                       ObservationGroupAxis('AZ', az, 'bin_edges'),
+                       ObservationGroupAxis('N_TELS', ntels, 'bin_values')]
+obs_groups = ObservationGroups(list_obs_group_axis)
+obs_table_grouped = obs_groups.group_observation_table(Observation_Table)
```

Comment:
Ok I didn’t undersand why the observation table didn’t have the ALT column… I thought that it was already implemented.
Whet do you think it’s the best? Not sure to understand the first point?
I don’t want to spend to much time on this but we need something that produce a suitable observation table if we want other people use the grouping...
Le 20 janv. 2016 à 12:01, Christoph Deil <notifications@github.com> a écrit :

> In checks/grouping.py:
> 
> > +from gammapy.data import ObservationGroups
> > +from gammapy.datasets import gammapy_extra
> > +import pylab as pt
> > +pt.ion()
> > +
> > +dir=str(gammapy_extra.dir)+'/datasets/hess-crab4'
> > +data_store = DataStore.from_dir(dir)
> > +Observation_Table=data_store.obs_table
> > +alt = Angle([0, 30, 60, 90], 'degree')
> > +az = Angle([-90, 90, 270], 'degree')
> > +ntels = np.array([3, 4])
> > +list_obs_group_axis = [ObservationGroupAxis('ALT', alt, 'bin_edges'),
> > +                       ObservationGroupAxis('AZ', az, 'bin_edges'),
> > +                       ObservationGroupAxis('N_TELS', ntels, 'bin_values')]
> > +obs_groups = ObservationGroups(list_obs_group_axis)
> > +obs_table_grouped = obs_groups.group_observation_table(Observation_Table)
> This error makes sense, no?
> 
> In your observation group specification, you say that you want to group by ALT.
> 
> But the observation list you give as input doesn't have an ALT column:
> 
> $ ftlist $GAMMAPY_EXTRA/datasets/hess-crab4/observations.fits.gz C
> HDU 2  
> 
>   Col  Name             Format[Units](Range)      Comment
>     1 OBS_ID             6A                   
> To test observation grouping code, you need an example observation table that has the parameters you care about (ALT, muon efficiency, ...).
> 
> There's a few options how you could proceed:
> 
> Produce an observation index table in the HAP-Fr scripts you have (using the format specified here), and then just take 10 or 100 rows and put them as an example file in gammapy-extra, and use that for tests. I don't think giving away some info what we observed in ~ 2004 is a concern.
> Update the Python scripts in this hess-crab4 folder to simulate a suitable observation table.
> Update the gammapy.datasets.make_test_observation_table function to use the agreed format and add the info you need. Generally the Gammapy obs handling code needs to be updated / improved, see e.g. here or here. I want to do this, but don't have the time. Probably we'll have to go in small steps / pull requests to improve what we have ...
> —
> Reply to this email directly or view it on GitHub.
>